selfie.c

```diff
@@ -1040,7 +1040,10 @@ void init_memory(uint64_t megabytes);
 uint64_t load_physical_memory(uint64_t* paddr);
 void     store_physical_memory(uint64_t* paddr, uint64_t data);
 
-uint64_t frame_for_page(uint64_t* table, uint64_t page);
+uint64_t root_page_table_index_for_page(uint64_t page);
+uint64_t leaf_page_table_index_for_page(uint64_t page);
+
+uint64_t frame_for_page(uint64_t* table, uint64_t* hypervisor_table, uint64_t page);
```

Comment:
I'd still prefer `parent_table` over `hypervisor_table`. The parent is the context that created the context of `table`...